src/dbally/collection.py

```diff
@@ -48,7 +46,7 @@ def __init__(
         self,
         name: str,
         view_selector: ViewSelector,
-        iql_generator: IQLGenerator,
```

Comment:
Okay, so here the idea is that rather than injecting your own IQLGenerator on collection level you can override method `get_iql_generator` on your view?

Reply:
Yes. It's a byproduct of other changes but I think it might make a lot of sense sense. This means that the IQLGenerator implementation is declared per view (or per view type), not per collection. So for example, you can have extended IQL for a particular data source (connected with a view that knows how to handle it).

Reply:
My little concern is that Python programmers are probably not used to getters and setters that much. What if we kept the argument but expected a callable?